### PR TITLE
[MOSIP-35101] created ansible script to provision RKE2 cluster, updated install nfs-csi config for default storage class and renamed istio install.yaml to sh

### PR DESCRIPTION
**Pull request by bhumi46**

### CODE REVIEW

File: k8-cluster/on-prem/rke2/ansible/README.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+1. **Clone the repository:**
+
+    ```bash
+    git clone https://github.com/mosip/k8s-infra.git
+    cd <repository-directory>
+    ```
+
+2. **Edit `hosts.ini`:**
+
+    Update the `hosts.ini` file with your node details.
```
Comment (syedsalman3753) [inline k8-cluster/on-prem/rke2/ansible/README.md:114]: 2. Update the `hosts.ini` file with your node details.

File: k8-cluster/on-prem/rke2/ansible/README.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+The configuration templates are located in the `rke2` directory and are used by the playbooks to configure each node type. The templates include:
+
+- `rke2-server-control-plane-primary.conf.template`: Used by the primary control plane.
+- `rke2-server-control-plane.subsequent.conf.template`: Used by subsequent control planes.
+- `rke2-agents.conf.template`: Used by agent nodes.
+- `rke2-etcd-agents.conf.template`: Used by etcd nodes.
+
+These templates contain placeholders that are dynamically replaced with actual values during the playbook execution.
+
+### Running the Playbook
```
Comment (syedsalman3753) [inline k8-cluster/on-prem/rke2/ansible/README.md:101]: Kubernetes setup via Ansible playbook

**Review by syedsalman3753 — CHANGES_REQUESTED** [review]

File: k8-cluster/on-prem/rke2/ansible/main.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    - name: Install RKE2
+      shell: |
+        curl -sfL https://get.rke2.io | INSTALL_RKE2_VERSION="{{ INSTALL_RKE2_VERSION }}" sh -
+
+# Include the control_plane_primary playbook
+- import_playbook: playbook/control-plane-primary.yaml
+
+# Include subsequent roles playbook
+- import_playbook: playbook/subsequent-roles.yaml
```
[the remark this answers is not in the document]
Comment (bhumi46) [reply k8-cluster/on-prem/rke2/ansible/main.yaml:25]: it was already added in subsequent playbook, please check @syedsalman3753

File: k8-cluster/on-prem/rke2/ansible/README.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+cluster_domain=<cluster name>
+# The path where RKE2 will be installed on each node.
+RKE2_PATH=/etc/rancher/rke2
+
+```
+
+
+### Update the `hosts.ini` File
+
+Update the `hosts.ini` file with the correct IP addresses, usernames, and paths to your private key files.
```
[the remark this answers is not in the document]
Comment (bhumi46) [reply k8-cluster/on-prem/rke2/ansible/README.md:57]: updated

File: k8-cluster/on-prem/rke2/ansible/README.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+- [Configuration Details](#configuration-details)
+
+## Prerequisites
+
+Before running the playbooks, ensure the following:
+
+- **Ansible:** Installed on local machine.
+- **Access to Nodes:** SSH access to all nodes (primary control plane, subsequent control planes, agents, etcd) with appropriate permissions.
+- **Private Key:** Available for SSH authentication.
+- **Supported OS:** Ubuntu or other Debian-based distributions.
```
[the remark this answers is not in the document]
Comment (bhumi46) [reply k8-cluster/on-prem/rke2/ansible/README.md:20]: updated

File: k8-cluster/on-prem/rke2/ansible/README.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
+# RKE2 Kubernetes Cluster Setup using Ansible
+
+This repository contains an Ansible playbook and associated configurations to set up an RKE2 Kubernetes cluster with a primary control plane, subsequent control plane nodes, agent nodes, and an etcd cluster. The playbooks are designed to install RKE2, configure nodes, and ensure the cluster is ready for use.
```
Comment (ckm007) [inline k8-cluster/on-prem/rke2/ansible/README.md:3]: please get Document approved by Keshav.

File: k8-cluster/on-prem/rke2/ansible/README.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+- [Templates](#templates)
+- [Kubernetes setup via Ansible playbook](#kubernetes-setup-via-Ansible-playbook)
+- [Configuration Details](#configuration-details)
+
+## Prerequisites
+
+Before running the playbooks, ensure the following:
+
+- Install Ansible on your local machine.
+- Ensure that nodes are accessible from your local machine for SSH authentication.
```
Comment (ckm007) [inline k8-cluster/on-prem/rke2/ansible/README.md:18]: add playbook to check:
1. availability of ssh pem key to access all the required nodes.
2. performs the ssh connection and confirms the accesibility to the VM's.
3. presence of user with sudo priviledge in all the required instances.

File: k8-cluster/on-prem/rke2/ansible/README.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+- Verify that the SSH private key for the nodes is available on your local machine.
+- Supported operating systems include Ubuntu 24.04 and other Debian-based distributions.
+
+
+## Inventory Setup
+
```
Comment (ckm007) [inline k8-cluster/on-prem/rke2/ansible/README.md:24]: follow .md .
add pointers *.

File: k8-cluster/on-prem/rke2/ansible/README.md [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+The `hosts.ini` file contains the configuration of your nodes. It is structured as follows:
+
+```
+# Primary control plane node. This is the main node where RKE2 will be installed first.
```
Comment (ckm007) [inline k8-cluster/on-prem/rke2/ansible/README.md:28]: Dont display the contents of hosts.ini.
Rather make hosts.ini self explanatery.

Add comment to define the default roles for each type.

File: k8-cluster/on-prem/rke2/ansible/README.md [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+# The version of RKE2 to install across the cluster.
+INSTALL_RKE2_VERSION=v1.28.9+rke2r1
+# The DNS domain name to use for the cluster.
+cluster_domain=<cluster name>
+# The path where RKE2 will be installed on each node.
+RKE2_PATH=/etc/rancher/rke2
+
+```
+
+- Update the `hosts.ini` file with the correct IP addresses, usernames, and paths to your private key files.
```
Comment (ckm007) [inline k8-cluster/on-prem/rke2/ansible/README.md:55]: Update hosts.ini with required details.

File: k8-cluster/on-prem/rke2/ansible/README.md [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+
+### Playbooks
+
+#### `main.yaml`
```
Comment (ckm007) [inline k8-cluster/on-prem/rke2/ansible/README.md:59]: add playbook details in the README inside playbook directory which explains the usage deeply.

Here we should provide the detailed steps for                         Cluster creation , configuration etc.

File: k8-cluster/on-prem/rke2/ansible/main.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (ckm007) [reply k8-cluster/on-prem/rke2/ansible/main.yaml:25]: playbook/primary-node.yaml

playbook/subsequent-node.yaml

File: k8-cluster/on-prem/rke2/ansible/playbook/control-plane-primary.yaml [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+          
+          cat "{{ RKE2_PATH }}/rke2.yaml" | sed "s/127.0.0.1/${INTERNAL_IP}/g" | sed "s/default/${CLUSTER_DOMAIN}/g" | tee -a $HOME_DIR/.kube/${CLUSTER_DOMAIN}-${SUFFIX_NAME}.yaml
+          
+          chown -R {{ ansible_user }}:{{ ansible_user }} $HOME_DIR/.kube/*
+          chmod -R 444 $HOME_DIR/.kube/*.yaml
+          chmod +x /bin/kubectl
+        fi
+
+      args:
+        executable: /bin/bash
```
Comment (ckm007) [inline k8-cluster/on-prem/rke2/ansible/playbook/control-plane-primary.yaml:55]: add a task to copy the kubeconfig file into the existing directory of host from where ansible command was executed.

File: k8-cluster/on-prem/rke2/ansible/ports.yaml [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+        proto: tcp
+
+    # - name: Prometheus metrics
+    #   community.general.ufw:
+    #     rule: allow
+    #     port: 9796
+    #     proto: tcp
+
+
+    # Nginx ingress controller (not sure if this is required)
```
Comment (ckm007) [inline k8-cluster/on-prem/rke2/ansible/ports.yaml:30]: add in document, for observation cluster creation uncomment this section.

File: k8-cluster/on-prem/rke2/rke2-server-control-plane-primary.conf.template [hunk 14]
```diff
@@ -1,7 +1,7 @@
 token: <configure-some-token-here>
 node-name: <node-name>
 node-ip: <node-internal-ip>
-node-label: 
+node-label:
   - shouldInstallIstioIngress=true
```
Comment (ckm007) [inline k8-cluster/on-prem/rke2/rke2-server-control-plane-primary.conf.template:5]: Do we still need this label shouldInstallIstioIngress=true ?

**Review by ckm007 — APPROVED** [review]

**Pull request merged**